packages/ckeditor5-dev-changelog/tests/generatechangelog.test.ts

```diff
@@ -0,0 +1,327 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { generateChangelog } from '../src/generatechangelog.js';
+import { getPackageJsons } from '../src/utils/getreleasepackagespkgjsons.js';
+import { getPackageJson } from '../src/utils/getpackagejson.js';
+import { getChangesetFilePaths } from '../src/utils/getchangesetfilepaths.js';
+import { getChangesetsParsed } from '../src/utils/getchangesetsparsed.js';
+import { getSectionsWithEntries } from '../src/utils/getsectionswithentries.js';
+import { getNewVersion } from '../src/utils/getnewversion.js';
+import { getSectionsToDisplay } from '../src/utils/getsectionstodisplay.js';
+import { getReleasedPackagesInfo } from '../src/utils/getreleasedpackagesinfo.js';
+import { modifyChangelog } from '../src/utils/modifychangelog.js';
+import { removeChangesetFiles } from '../src/utils/removechangesetfiles.js';
+import { logInfo } from '../src/utils/loginfo.js';
+import { logChangelogFiles } from '../src/utils/logchangelogfiles.js';
+import { getRepositoryUrl } from '../src/utils/external/getrepositoryurl.js';
+import { SECTIONS } from '../src/constants.js';
+
+vi.mock( '../src/utils/getreleasepackagespkgjsons.js' );
+vi.mock( '../src/utils/getpackagejson.js' );
+vi.mock( '../src/utils/getchangesetfilepaths.js' );
+vi.mock( '../src/utils/getchangesetsparsed.js' );
+vi.mock( '../src/utils/getsectionswithentries.js' );
+vi.mock( '../src/utils/getnewversion.js' );
+vi.mock( '../src/utils/getsectionstodisplay.js' );
+vi.mock( '../src/utils/getreleasedpackagesinfo.js' );
+vi.mock( '../src/utils/modifychangelog.js' );
+vi.mock( '../src/utils/removechangesetfiles.js' );
+vi.mock( '../src/utils/loginfo.js' );
+vi.mock( '../src/utils/logchangelogfiles.js' );
+vi.mock( '../src/utils/external/getrepositoryurl.js' );
+vi.mock( 'chalk', () => ( {
+	default: {
+		yellow: ( text: string ) => text,
+		green: ( text: string ) => text,
+		cyan: ( text: string ) => text
+	}
+} ) );
+
+describe( 'generateChangelog()', () => {
+	const defaultOptions = {
+		cwd: '/home/ckeditor',
+		packagesDirectory: 'packages',
+		transformScope: ( name: string ) => ( {
+			displayName: name,
+			npmUrl: `https://www.npmjs.com/package/${ name }`
+		} ),
+		date: '2024-03-26' as const
+	};
+
+	beforeEach( () => {
+		vi.clearAllMocks();
```

Comment:
It seems that `vi.clearAllMocks()` is not needed, as we have `test.restoreMocks = true` in `vitest.config.ts`.

Reply:
Fixed in [6eb3e89](https://github.com/ckeditor/ckeditor5-dev/pull/1090/commits/6eb3e89b0561b626d5b5b5b8e41e7168b0a7096f).